### PR TITLE
[Core] Handle cpu == 0 more gracefully for the dashboard agent report

### LINKED ISSUE #34895: [dashboard] Dashboard error for cluster with CPU <1 with division by zero
## What Happened
Attempting to start a Ray cluster on version 2.4.0 with CPU <1, a fractional number such as 0.5 creates a division by zero, immediately crashing the server.

```
2023-04-29 00:01:45,428	ERROR reporter_agent.py:1036 -- Error publishing node physical stats.
Traceback (most recent call last):
  File "/home/ray/anaconda3/lib/python3.7/site-packages/ray/dashboard/modules/reporter/reporter_agent.py", line 1019, in _perform_iteration
    stats = self._get_all_stats()
  File "/home/ray/anaconda3/lib/python3.7/site-packages/ray/dashboard/modules/reporter/reporter_agent.py", line 605, in _get_all_stats
    "loadAvg": self._get_load_avg(),
  File "/home/ray/anaconda3/lib/python3.7/site-packages/ray/dashboard/modules/reporter/reporter_agent.py", line 572, in _get_load_avg
    per_cpu_load = tuple((round(x / self._cpu_counts[0], 2) for x in load))
  File "/home/ray/anaconda3/lib/python3.7/site-packages/ray/dashboard/modules/reporter/reporter_agent.py", line 572, in <genexpr>
    per_cpu_load = tuple((round(x / self._cpu_counts[0], 2) for x in load))
ZeroDivisionError: float division by zero
2023-04-29 00:01:46,330	INFO web_log.py:206 -- 127.0.0.1 [29/Apr/2023:07:01:46 +0000] "GET /api/local_raylet_healthz HTTP/1.1" 200 163 "-" "Wget/1.20.3 (linux-gnu)"
2023-04-29 00:01:47,022	WARNING agent.py:517 -- Exiting with SIGTERM immediately...
```

## What's Expected

Clusters should allow fractional CPU <1. If this setup is not suggested create a warning but do not crash the entire server.

### Versions / Dependencies

Ray 2.4.0
Kuberay 0.5.0 (via helm chart)

### Reproduction script

Create a kuberay cluster where the headnode CPU is <1, some fractional number like 0.5

### Issue Severity

Medium: It is a significant difficulty but I can work around it.

Leaving as medium because I think this could break some setups upgrading an existing <1 CPU setup. This also took going into the dashboard to diagnose the issue.

### PR DESCRIPTION
**Pull request by rkooo567**
## Why are these changes needed?

Seems like when logical_cpus == 0 (idk when this can happen, but the user raises a bug), the load_avg calculation doesn't work because of division by zero error. This PR fixes the issue. I also verified if cpus > 0, the issue from the tagged link doesn't happen. 

## Related issue number

Closes https://github.com/ray-project/ray/issues/34895

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: dashboard/modules/reporter/reporter_agent.py [hunk 1]
```diff
@@ -595,12 +595,16 @@ def _get_agent(self):
...
 
     def _get_load_avg(self):
+        logger.error("SANG-TODO")
         if sys.platform == "win32":
             cpu_percent = psutil.cpu_percent()
             load = (cpu_percent, cpu_percent, cpu_percent)
         else:
             load = os.getloadavg()
-        per_cpu_load = tuple((round(x / self._cpu_counts[0], 2) for x in load))
+        if self._cpu_counts[0] > 0:
```
Comment (chaowanggg) [inline dashboard/modules/reporter/reporter_agent.py:604]: Is there a possibility that self._cpu_counts is None?

**Review by chaowanggg — APPROVED** [review]
lgtm!

File: dashboard/modules/reporter/reporter_agent.py [hunk 2]
```diff
@@ -595,12 +595,16 @@ def _get_agent(self):
         )
 
     def _get_load_avg(self):
+        logger.error("SANG-TODO")
```
Comment (chaowanggg) [inline dashboard/modules/reporter/reporter_agent.py:598]: Why do you need a TODO here?

**Review by rickyyx — APPROVED** [review]

File: dashboard/modules/reporter/reporter_agent.py (hunk 2 above)
Comment (rkooo567) [reply dashboard/modules/reporter/reporter_agent.py]: lol fixed

**Pull request merged**